consensus/wal_test.go

```diff
@@ -16,6 +19,46 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestWALTruncate(t *testing.T) {
+	walDir, err := ioutil.TempDir("", "wal")
+	if err != nil {
+		panic(fmt.Errorf("failed to create temp WAL file: %v", err))
+	}
+
+	walFile := filepath.Join(walDir, "wal")
+
+	wal, err := NewWAL(walFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wal.group.SetHeadSizeLimit(4096) //cause fragment
+	wal.Start()
+
+	err = WALGenerateNBlocks(wal.group, 60) //60 block's size nearly 70K
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(5 * time.Second) //wait groupCheckDuration, make sure RotateFile run
+
+	wal.group.Flush()
```

Comment:
`wal.Group().Flush()`

generally you should not access private fields / methods (even though you can) - sign that there's something wrong with your API